src/openpgp.js

```diff
@@ -169,17 +169,16 @@ export function decryptKey({ privateKey, passphrase }) {
 export function encrypt({ data, publicKeys, privateKeys, passwords, filename, armor=true }) {
   checkData(data); publicKeys = toArray(publicKeys); privateKeys = toArray(privateKeys); passwords = toArray(passwords);
 
-  if (asyncProxy) { // use web worker if available
+  if (!util.getWebCrypto() && asyncProxy) { // use web worker if web crypto apis are not supported
     return asyncProxy.delegate('encrypt', { data, publicKeys, privateKeys, passwords, filename, armor });
   }
 
-  return execute(() => {
+  let message = createMessage(data, filename);
+  if (privateKeys) { // sign the message only if private keys are specified
+    message = message.sign(privateKeys);
+  }
```

Comment:
TODO: Oops. Should wrap this inside a promise so that errors are thrown up asynchronously.